utils/src/main/kotlin/ProcessCapture.kt

```diff
@@ -38,6 +38,17 @@ class ProcessCapture(workingDir: File?, environment: Map<String, String>, vararg
     companion object {
         private const val MAX_LOG_LINES = 20
         private const val MAX_LOG_LINES_MESSAGE = "(Above output is limited to $MAX_LOG_LINES lines.)"
+
+        private fun logLinesLimited(prefix: String, lines: Sequence<String>, limit: Int, logLine: (String) -> Unit) {
+            var count = 0
+            val i = lines.iterator()
+
+            while (i.hasNext() && count++ < limit) {
+                logLine("$prefix: ${i.next()}")
+            }
+
+            if (i.hasNext()) logLine(MAX_LOG_LINES_MESSAGE)
```

Comment:
What about using `forEachIndexed`:
```kotlin
            lines.forEachIndexed { index, line ->
                if (index >= limit) {
                    logLine(MAX_LOG_LINES_MESSAGE)
                    return@forEachIndexed
                }

                logLine("$prefix: $line")
            }
```

Reply:
Even shorter:
```kotlin
            val chunkIterator = lines.chunked(limit).iterator()
            chunkIterator.next().forEach { logLine("$prefix: $it") }
            if (chunkIterator.hasNext()) { logLine(MAX_LOG_LINES_MESSAGE) }
```

Reply:
I was considering the latter, but I wasn't sure whether the sequence of lists that `chunked()` returns is really evaluated lazily. For long output (like from git-repo clones) I'd like to really only process the output lines that are being logged. I'll investigate a bit more.

Reply:
Done (chunked version).

Reply:
I does create the `next()` element only when required, but after only shortly looking at the code it wasn't clear to me if calling `hasNext()` also creates the next element already. Did you find out?

Reply:
No, not certainly...

Reply:
I just made a test, calling `chunkIterator.hasNext()` already creates the elements of the next chunk which is what you want to avoid.

Reply:
Crap. But then this probably also hold true for my initial implementation with directly using an iterator, or?

Reply:
If so, in my initial implementation only the next line would be unnecessarily read. In the chunked approach, only the next chunk, i.e. the next 20 lines, would be unnecessarily read. Still not as bad as reading all of the (potentially huge) rest of the output. So which solution do you prefer? I'd be fine with the chucked approach for its elegance.

Reply:
For reference: https://try.kotlinlang.org/#/UserProjects/4jkelkle8g76o04vvu335beq4b/klk2qpphgr7qsi5t8buemo6cmm

I'm fine with every solution that does not manually manage a `count` variable ;)
My first proposal also reads only one more line than shall be printed, but I think that cannot be avoided because we have to check if there is one more line.

Reply:
Then I'd refer to stick with the chucked approach as it's slightly shorter and avoids `return@` which I always find hard to read.

PS: Nice code snippet on *try.kotlinlang.org*, really cool to be able to share small test code that way!

Reply:
Let me actually add a code comment, though!